tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -108,88 +108,103 @@ object SQLPlanParser extends Logging {
+          case "FlatMapGroupsInPandas" =>
+            FlatMapGroupsInPandasExecParser(node, checker, sqlID).parse
+          case "Generate" =>
+            GenerateExecParser(node, checker, sqlID).parse
+          case "GlobalLimit" =>
+            GlobalLimitExecParser(node, checker, sqlID).parse
+          case "HashAggregate" =>
+            HashAggregateExecParser(node, checker, sqlID, app).parse
+          case "LocalLimit" =>
+            LocalLimitExecParser(node, checker, sqlID).parse
+          case "InMemoryTableScan" =>
+            InMemoryTableScanExecParser(node, checker, sqlID).parse
+          case i if (i.contains("InsertIntoHadoopFsRelationCommand") ||
+            i == "DataWritingCommandExec") =>
+            DataWritingCommandExecParser(node, checker, sqlID).parse
+          case "MapInPandas" =>
+            MapInPandasExecParser(node, checker, sqlID).parse
+          case "ObjectHashAggregate" =>
+            ObjectHashAggregateExecParser(node, checker, sqlID, app).parse
+          case "Project" =>
+            ProjectExecParser(node, checker, sqlID).parse
+          case "Range" =>
+            RangeExecParser(node, checker, sqlID).parse
+          case "Sample" =>
+            SampleExecParser(node, checker, sqlID).parse
+          case "ShuffledHashJoin" =>
+            ShuffledHashJoinExecParser(node, checker, sqlID, app).parse
+          case "Sort" =>
+            SortExecParser(node, checker, sqlID).parse
+          case s if (s.startsWith("Scan")) =>
+            FileSourceScanExecParser(node, checker, sqlID, app).parse
+          case "SortAggregate" =>
+            SortAggregateExecParser(node, checker, sqlID).parse
+          case "SortMergeJoin" =>
+            SortMergeJoinExecParser(node, checker, sqlID).parse
+          case "SubqueryBroadcast" =>
+            SubqueryBroadcastExecParser(node, checker, sqlID, app).parse
+          case "TakeOrderedAndProject" =>
+            TakeOrderedAndProjectExecParser(node, checker, sqlID).parse
+          case "Union" =>
+            UnionExecParser(node, checker, sqlID).parse
+          case "Window" =>
+            WindowExecParser(node, checker, sqlID).parse
+          case "WindowInPandas" =>
+            WindowInPandasExecParser(node, checker, sqlID).parse
+          case _ =>
+            new ExecInfo(sqlID, node.name, expr = "", 1, duration = None, node.id,
+              isSupported = false, None)
+        }
+      } catch {
+        // Error parsing expression could trigger an exception. If the exception is not handled,
+        // the application will be skipped. We need to suppress exceptions here to avoid
+        // sacrificing the entire app analysis.
+        // Note that:
+        //  - The exec will be considered unsupported.
+        //  - No need to add the SQL to the failed SQLs, because this will cause the app to be
+        //    labeled as "Not Applicable" which is not preferred at this point.
+        case e: Throwable =>
```

Comment:
```suggestion
        case NonFatal(e) =>
```

Reply:
I was not aware of `NonFatal` in scala.
Thanks @gerashegalov !